test/services/optIn/OptInPropositionSpec.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.optIn
+
+import models.incomeSourceDetails.TaxYear
+import models.itsaStatus.ITSAStatus.{Annual, ITSAStatus, Mandated, NoStatus, Voluntary}
+import services.optIn.OptInPropositionSpec.TestData
+import services.optIn.core._
+import testUtils.UnitSpec
+
+object OptInPropositionSpec {
+  case class TestData(currentTaxYearStatus: ITSAStatus, nextTaxYearStatus: ITSAStatus, currentTaxYear: TaxYear, intent: String, offered: Seq[String])
+}
+
+class OptInPropositionSpec extends UnitSpec {
+
+  val forYearEnd = 2023
+  private val currentTaxYear = TaxYear.forYearEnd(forYearEnd)
+
+  private val validTestCases = List(
+    TestData(Mandated, Annual, currentTaxYear, "NY", Seq("NY")),
```

Comment:
Ideally, this test data should be generated from the spreadsheet, as it is in `OptOutPropositionSpec`.

Reply:
can be done in separate ticket

Reply:
OK, let's make another sub-task under the current ticket.